hal/src/nRF52840/exflash_hal.c

```diff
@@ -574,3 +581,33 @@ int hal_exflash_special_command(hal_exflash_special_sector_t sp, hal_exflash_com
     hal_exflash_unlock();
     return ret;
 }
+
+int hal_exflash_sleep(bool sleep, void* reserved) {
+    hal_exflash_lock();
```

Comment:
We should acquire the lock if entering sleep mode is requested, and release it after exiting sleep mode.

Reply:
Not very clear why we should lock the external flash so long time. And in stop mode, we also stop the scheduler.